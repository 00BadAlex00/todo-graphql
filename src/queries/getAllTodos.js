import { gql } from '@apollo/client'

import { TODO_FIELDS } from '../fragments'

const ALL_TODOS = gql`
  ${TODO_FIELDS}
  query allTodos {
    allTodos {
      ...TodoFields
    }
  }
`
export default ALL_TODOS;
