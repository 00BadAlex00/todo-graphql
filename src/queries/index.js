import ALL_TODOS from './getAllTodos'

export {
  ALL_TODOS,
}
