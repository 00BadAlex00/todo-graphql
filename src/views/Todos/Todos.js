import { useQuery } from '@apollo/client'

import { ALL_TODOS } from '../../queries'
import Error from '../shared/Error'
import Loading from '../shared/Loading'
import TodoItem from './TodoItem'

const Todos = () => {
  const { loading, error, data } = useQuery(ALL_TODOS)

  if (loading) return <Loading />
  if (error) return <Error />
  return (
    <ul className="todo-list">
      {data.allTodos.map((todo) => (
        <TodoItem todo={todo} key={todo.id} />
      ))}
    </ul>
  )
}

export default Todos
