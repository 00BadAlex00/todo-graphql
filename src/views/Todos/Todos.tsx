import { useQuery } from '@apollo/client'

import { ALL_TODOS } from '../../queries'
import Error from '../shared/Error'
import Loading from '../shared/Loading'
import TodoItem from './TodoItem'

interface TodosData {
  allTodos: Todo[];
}

const Todos = () => {
  const { loading, error, data } = useQuery<TodosData>(ALL_TODOS)

  if (loading) return <Loading />
  if (error) return <Error />
  return (
    <ul className="todo-list">
      {data?.allTodos.map((todo) => (
        <TodoItem key={todo.id} {...todo} />
      ))}
    </ul>
  )
}

export default Todos
