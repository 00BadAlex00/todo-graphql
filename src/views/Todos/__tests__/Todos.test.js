import React from "react";
import { MockedProvider } from "@apollo/client/testing";
import renderer, { act } from "react-test-renderer";
import wait from "waait";

import { ALL_TODOS } from "../../../queries";
import Todos from "../Todos";

describe('component "Todos"', () => {
  const getComponent = (mocks = []) => (
    renderer.create(
      <MockedProvider
        mocks={mocks}
        addTypename={false}
        defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}
      >
        <Todos />
      </MockedProvider>
    )
  )

  it("matches snapshot with loading", () => {
    const mocks = [{
      request: {
        query: ALL_TODOS,
      },
    }];
    const component = getComponent(mocks)

    expect(component).toMatchSnapshot()
  });

  it("matches snapshot with error", async () => {
    const mocks = [{
      request: {
        query: ALL_TODOS,
      },
      error: new Error('An error occurred'),
    }];

    let component
    act(() => component = getComponent(mocks))
    await act(() => wait(0));

    expect(component).toMatchSnapshot()
  });

  it("matches snapshot with data", async () => {
    const mocks = [{
      request: {
        query: ALL_TODOS,
      },
      result: {
        data: {
          allTodos: [{
            completed: true,
            id: '1',
            title: 'title',
            // __typename: "Todo"
          }]
        }
      }
    }];

    let component
    act(() => component = getComponent(mocks))
    await act(() => wait(0));

    expect(component).toMatchSnapshot()
  });
});
