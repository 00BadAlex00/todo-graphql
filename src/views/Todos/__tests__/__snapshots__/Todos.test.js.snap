// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component "Todos" matches snapshot with data 1`] = `
<ul
  className="todo-list"
>
  <li
    className="todo completed"
  >
    <div
      className="view"
    >
      <input
        checked={true}
        className="toggle"
        disabled={false}
        onChange={[Function]}
        type="checkbox"
      />
      <label>
        title
      </label>
      <button
        className="destroy"
        onClick={[Function]}
      />
    </div>
  </li>
</ul>
`;

exports[`component "Todos" matches snapshot with error 1`] = `
<p
  className="loading-error"
>
  Error :(
</p>
`;

exports[`component "Todos" matches snapshot with loading 1`] = `
<div
  className="todos-container"
>
  <p
    className="loading"
  >
    Loading...
  </p>
</div>
`;
